zio-http/src/main/scala/zhttp/http/middleware/AuthMiddleware.scala

```diff
@@ -0,0 +1,88 @@
+package zhttp.http.middleware
+
+import io.netty.handler.codec.http.HttpHeaderNames
+import pdi.jwt._
+import pdi.jwt.algorithms.JwtHmacAlgorithm
+import zhttp.http.HeaderExtension.BasicSchemeName
+import zhttp.http.{Header, HeaderExtension, HttpApp, HttpError}
+import zio.ZIO
+
+/**
+ * Authentication Middlewares for HttpApp.
+ */
+sealed trait AuthMiddleware[-R, +E] { self =>
```

Comment:
Can we not use `HttpMiddleware` for authentication?

Reply:
We can but the orElse operator won't work as expected